thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/alert/feed/UnionAnomalyFeed.java

```diff
@@ -0,0 +1,106 @@
+package com.linkedin.thirdeye.alert.feed;
+
+import com.linkedin.thirdeye.alert.commons.AnomalyFeedConfig;
+import com.linkedin.thirdeye.alert.commons.AnomalyFetcherConfig;
+import com.linkedin.thirdeye.alert.commons.AnomalyFetcherFactory;
+import com.linkedin.thirdeye.alert.commons.AnomalySource;
+import com.linkedin.thirdeye.alert.fetcher.AnomalyFetcher;
+import com.linkedin.thirdeye.anomaly.alert.util.AlertFilterHelper;
+import com.linkedin.thirdeye.datalayer.bao.AlertSnapshotManager;
+import com.linkedin.thirdeye.datalayer.dto.AlertSnapshotDTO;
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import com.linkedin.thirdeye.datasource.DAORegistry;
+import com.linkedin.thirdeye.detector.email.filter.AlertFilter;
+import com.linkedin.thirdeye.detector.email.filter.AlertFilterFactory;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.List;
+import java.util.Set;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class UnionAnomalyFeed implements AnomalyFeed {
+  private static final Logger LOG = LoggerFactory.getLogger(UnionAnomalyFeed.class);
+
+  private AlertSnapshotManager alertSnapshotDAO;
+  private AlertFilterFactory alertFilterFactory;
+
+  private List<AnomalyFetcher> anomalyFetchers;
+  private List<AlertFilter> alertFilters;
+  private AlertSnapshotDTO alertSnapshot;
+  private DateTime alertTime;
+
+  public UnionAnomalyFeed(){
+    this.anomalyFetchers = new ArrayList<>();
+    this.alertFilters = new ArrayList<>();
+    this.alertTime = DateTime.now();
+  }
+
+  @Override
+  public void init(AlertFilterFactory alertFilterFactory, AnomalyFeedConfig anomalyFeedConfig) {
+    this.alertSnapshotDAO = DAORegistry.getInstance().getAlertSnapshotDAO();
+    this.alertFilterFactory = alertFilterFactory;
+    AnomalySource anomalySourceType = anomalyFeedConfig.getAnomalySourceType();
+    String anomalySource = anomalyFeedConfig.getAnomalySource();
+    if (anomalyFeedConfig.getAlertSnapshotId() != null) {
+      alertSnapshot = alertSnapshotDAO.findById(anomalyFeedConfig.getAlertSnapshotId());
```

Comment:
Where is alertSnapshot created for the first time?

Reply:
It is created in AlertTaskRunnerV2. It should be the place where the alertSnapshot is created. Otherwise, the alertConfig has no idea about the new generated alertSnapshot.